float32 curr_t